guacamole-ext/src/main/java/org/apache/guacamole/net/auth/Connection.java

```diff
@@ -125,5 +125,17 @@
      *     connection.
      */
     public Set<String> getSharingProfileIdentifiers() throws GuacamoleException;
+    
+    /**
+     * Get the GuacamoleProxyConfiguration for this connection.
+     * 
+     * @return
+     *     The GuacamoleProxyConfiguration for this connection.
+     * 
+     * @throws GuacamoleException 
+     *     If configuration information cannot be retrieved or parsed.
```

Comment:
Implementations will not necessarily be retrieving or parsing this information. We can't assume that `GuacamoleException` will be thrown only due to failures from function calls made against an `Environment`.

Any thoughts on not throwing `GuacamoleException`? `getConfiguration()` does not do so.

Reply:
Pulled this out.